challenger.go

```diff
@@ -79,3 +214,39 @@ func (l *LndChallenger) NewChallenge(price int64) (string, lntypes.Hash, error)
 
 	return response.PaymentRequest, paymentHash, nil
 }
+
+// VerifyInvoiceStatus checks that an invoice identified by a payment
+// hash has the desired status. To make sure we don't fail while the
+// invoice update is still on its way, we try several times until either
+// the desired status is set or the given timeout is reached.
+//
+// NOTE: This is part of the auth.InvoiceChecker interface.
+func (l *LndChallenger) VerifyInvoiceStatus(hash lntypes.Hash,
+	state lnrpc.Invoice_InvoiceState, timeout time.Duration) error {
+
+	exitTimer := time.After(timeout)
+	for {
+		l.invoicesMtx.Lock()
+		invoiceState, ok := l.invoiceStates[hash]
+		l.invoicesMtx.Unlock()
+
+		// Exit early if we have the invoice with the correct status.
+		if ok && invoiceState == state {
+			log.Debugf("Invoice for hash %v has status: %v",
+				hash, invoiceState)
+			return nil
+		}
+
+		<-time.After(invoicePollInterval)
```

Comment:
Should also check quit here.

Reply:
If we go with what I described above, we can eliminate the timer all together.